packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -1201,53 +1171,16 @@ Quad _transformViewport(Matrix4 matrix, Rect viewport) {
   );
 }
 
-// Find the axis aligned bounding box for the rect rotated about its center by
-// the given amount.
-Quad _getAxisAlignedBoundingBoxWithRotation(Rect rect, double rotation) {
-  final Matrix4 rotationMatrix = Matrix4.identity()
-      ..translate(rect.size.width / 2, rect.size.height / 2)
-      ..rotateZ(rotation)
-      ..translate(-rect.size.width / 2, -rect.size.height / 2);
-  final Quad boundariesRotated = Quad.points(
-    rotationMatrix.transform3(Vector3(rect.left, rect.top, 0.0)),
-    rotationMatrix.transform3(Vector3(rect.right, rect.top, 0.0)),
-    rotationMatrix.transform3(Vector3(rect.right, rect.bottom, 0.0)),
-    rotationMatrix.transform3(Vector3(rect.left, rect.bottom, 0.0)),
-  );
-  return InteractiveViewer.getAxisAlignedBoundingBox(boundariesRotated);
-}
-
-// Return the amount that viewport lies outside of boundary. If the viewport
-// is completely contained within the boundary (inclusively), then returns
-// Offset.zero.
-Offset _exceedsBy(Quad boundary, Quad viewport) {
-  final List<Vector3> viewportPoints = <Vector3>[
-    viewport.point0, viewport.point1, viewport.point2, viewport.point3,
-  ];
-  Offset largestExcess = Offset.zero;
-  for (final Vector3 point in viewportPoints) {
-    final Vector3 pointInside = InteractiveViewer.getNearestPointInside(point, boundary);
-    final Offset excess = Offset(
-      pointInside.x - point.x,
-      pointInside.y - point.y,
-    );
-    if (excess.dx.abs() > largestExcess.dx.abs()) {
-      largestExcess = Offset(excess.dx, largestExcess.dy);
-    }
-    if (excess.dy.abs() > largestExcess.dy.abs()) {
-      largestExcess = Offset(largestExcess.dx, excess.dy);
-    }
-  }
 
-  return _round(largestExcess);
-}
+const double _kTolerance = 100000000000.0;
```

Comment:
There are a bunch of places where I had to use rounding in order to get numbers to match.  Do we have a standardized way of handling stuff like this somewhere?

Reply:
> get numbers to match

You mean checking for floating-point equality/comparison? As far as I can tell we just check if 2 floating numbers are within an `epsilon` range, eg:

https://github.com/flutter/flutter/blob/02d441ea55b328133c266991f43b0a1148edb63f/packages/flutter/lib/src/physics/utils.dart#L5-L21